sweep/sweeper.go

```diff
@@ -1,24 +1,59 @@
 package sweep
 
 import (
+	"errors"
+	"fmt"
 	"github.com/btcsuite/btcd/blockchain"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
 	"github.com/btcsuite/btcd/txscript"
 	"github.com/btcsuite/btcd/wire"
 	"github.com/btcsuite/btcutil"
+	"github.com/lightningnetwork/lnd/chainntnfs"
 	"github.com/lightningnetwork/lnd/lnwallet"
+	"github.com/lightningnetwork/lnd/ticker"
+	"sync"
+	"sync/atomic"
+	"time"
 )
 
-// UtxoSweeper provides the functionality to generate sweep txes. The plan is
-// to extend UtxoSweeper in the future to also manage the actual sweeping
-// process by itself.
+var (
+	// DefaultSweepCheckInterval specifies the time intervals between checks
+	// for inputs to sweep.
+	DefaultSweepCheckInterval = time.Second
```

Comment:
Perhaps we should make this multiple seconds? Possibly 10 or so? A larger value allows us to batch more inputs. Consider the start up case, and also the case where a block is mined, and each client has a distinct block epoch subscription and may do actions like write to disk or w/e being requesting the sweeper's service.

Reply:
Yes, so the batch window is currently DefaultSweepCheckInterval * DefaultBatchWindowSize.